### ISSUE TITLE
multiple projections for transcript endpoint

### ISSUE THREAD

**Issue by inodb** [genome-nexus/genome-nexus#197]
transcript endpoint is getting p filled with pfam, exon and utr info. Might be good to do a summary/detailed projection

**Comment by jjgao** [comment]
@inodb @pieterlukasse @sandertan can we prioritize this one before the new PFAM etc info is used in production? 

I am thinking of generalizing it to pull annotation on a protein instead of a variant.

Maybe we can have two annotation endpoints: `variant-annotation` and `protein-annotation`

`variant-annotation` will return annotations relevant to a variant. It's basically our current `annotation` endpoint.

`protein-annotation` will return annotations relevant to a protein. It can return all kinds of non-variant-specific annotations including pfam, exon, PTM, conservation, etc. It will be the basis of the next multi-track mutation diagram.

**Comment by jjgao** [comment]
We can also keep the data format generic, e.g.:

```
[
  {
    "transcriptId": "ENST000xxxx",
    "geneId": "ENSG000xxxx",
    "hugoSymbols": [
      "MTRNR2L7"
    ],
    "proteinId": "ENSP0xxxx",
    "proteinLength": 240,
	"annotations": {
		"exon":
		[
			{
				"proteinStart": 1,
				"proteinEnd": 50,
				"data":
				{
					"rank": 1,
					"strand": -1
				}
			},
			{
				"proteinStart": 51,
				"proteinEnd": 240,
				"data":
				{
					"rank": 2,
					"strand": -1
				}
			}
		],
		"pfam":
		[
			{
				"proteinStart": 1,
				"proteinEnd": 24,
				"data":
				{
					"pfamId": "PFxxxx",
					
				}
			}
		],
		"Phosphorylation":
		[
			{
				"proteinStart": 101,
				"proteinEnd": 101,
				"data":
				{
					"type": "phosphoserine"
				}
			}
		]
	}
]
```

**Comment by sandertan** [comment]
Hi @jjgao , I'll let @pieterlukasse respond on this design. A minor comment on the above contents:

'Exons' are part of the transcript, not the protein. They are in basepairs on the genome. Same goes for UTRs.

**Comment by jjgao** [comment]
@sandertan agreed that UTRs probably do not belong there. I think exon-exon breakpoints can be annotated on protein sequence -- actually, that's what was visualized in the Fusion diagram. I am also ok to keep the exon data in the transcript endpoint as long as we move the Pfam data.

**Comment by pieterlukasse** [comment]
@jjgao looks good to me, although we need to be able to query the endpoint using transcript id and we need also genomeStart and genomeEnd (bp positions of the start /end ) for exons. The current Fusion diagram uses the bp positions and UTR length to determine where to draw the pfam domains. E.g. if pfam domain has proteinStart=28, it will be drawn at postion `3*28+length(5'UTR)`.

**Comment by jjgao** [comment]
@pieterlukasse @sandertan I see. Do you think it's possible to change your code and use protein locations (the exon-exon breaking points and pfam domains) to draw the diagram. Then you may not need the UTR data? Would that simplify things?

**Comment by sandertan** [comment]
I think it might complicate things. Because we currently draw the full transcript, which starts with the first exon, containing the 5'UTR. Also, the 3'UTR at the end is drawn, because it is contained in the last exons. 

For the exon-exon breakpoints that fall within the protein, it's possible to get a mapping, but there's no mapping from exon to protein for locations that are UTR.

Also, I think including the UTR is a common way of drawing a transcript and its exons.

**Comment by onursumer** [comment]
@jjgao Yes, we are using this endpoint: http://genomenexus.org/swagger-ui.html#!/ensembl45controller/fetchCanonicalEnsemblTranscriptByHugoSymbolGET

**Comment by jjgao** [comment]
@onursumer thanks.

**Comment by jjgao** [comment]
@sandertan @pieterlukasse I thought it a little more. The fusion diagram and the mutation diagram are very similar. Potentially we can support more tracks in both diagrams. And ideally, we can reuse components across the two diagrams (e.g. we could use the same protein part and Fusion diagram would just add fusion alignments, and if we want to add new annotations, e.g. PTM sites, we could easily add to both on protein level).

With that being said, I am not sure if it's a good idea to use genomic locations in the fusion diagram. If UTR is the only reason, I am not sure if it is worth it -- we would have to maintain two sets of code for very similar functionality. If the fusion breakpoint is in UTR, one option would just connecting to the first coding exon (this is similar to a breakpoint in intron).

And if you look at [the figure from the paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5461196/figure/F4/), there are no UTRs at all. In fact, the whole diagram was based on protein sequence if you look at the position labels. I don't really see transcript as important as protein in the figure.

@sandertan @pieterlukasse would you please discuss more and see if you can reconsider? Thanks!

**Comment by jjgao** [comment]
... and sorry for being stubborn on this. I think it's important for both cBioPortal and Genome Nexus in terms of code maintainability.
